src/utils.ts

```diff
@@ -0,0 +1,67 @@
+import * as core from "@actions/core";
+import { read } from "@1password/op-js";
+import {
+	authErr,
+	envConnectHost,
+	envConnectToken,
+	envServiceAccountToken,
+	envManagedVariables,
+} from "./constants";
+
+export const semverToInt = (input: string): string =>
+	input
+		.split(".")
+		.map((n) => n.padStart(2, "0"))
+		.join("");
+
+export const validateAuth = (): void => {
+	let authType = "Connect";
+	if (!process.env[envConnectHost] || !process.env[envConnectToken]) {
+		if (!process.env[envServiceAccountToken]) {
+			throw new Error(authErr);
+		}
+		authType = "Service account";
+	}
```

Comment:
Was just testing out some local changes to this section of the code; we can simplify it quite a bit with the following changes:

1. Don't declare `authType`/default it to "Connect".
2. Create some boolean checks:
  ```ts
  const isConnect = process.env[envConnectHost] && process.env[envConnectToken];
  const isServiceAccount = process.env[envServiceAccountToken];
  ```
3. Use these booleans to check for issues with the environment variables:
  ```ts
  if (isConnect && isServiceAccount) {
    throw new Error("Environment variables were passed for both Connect and Service Accounts; you must choose either one or the other, but not both.");
  }
  if (!isConnect && !isServiceAccount) {
    throw new Error(authErr);
  }
  ```
4. If there are no issues with the env vars, then we can set the authType.
  ```ts
  const authType = isConnect ? "Connect" : "Service account";
  ```

Curious to hear if these changes make sense to you, especially the additional error handling that I added if someone provides all three env vars (i.e. two for Connect, one for Service Accounts).

Reply:
+1

Reply:
This definitely looks way better and cleaner.

The only part of the code that I'm slightly hesitant to include is the following:
```ts
if (isConnect && isServiceAccount) {
  throw new Error("Environment variables were passed for both Connect and Service Accounts; you must choose either one or the other, but not both.");
}
```

The reason is that the CLI can handle this edge case by having Connect take priority. We count on this behavior in multiple places and we also [let the users know](https://developer.1password.com/docs/ci-cd/github-actions) that if this happens, then Connect takes priority.

It definitely feels better to stop the user from continuing when having two configurations since it can get confusing, however it may lead to a slight anti-pattern to what we do in other integrations. 🤔 
This can also be looked the other way around as in we should update all integrations to just stop if both are configured. 😄

Reply:
After some more discussions, we've decided that we will not add this additional enhancement on top of the CLI. Instead, we will print a warning to the user in the pipeline that both are configured and that Connect will take priority.
```ts
if (isConnect && isServiceAccount) {
	core.warning(
		"WARNING: Both service account and Connect credentials are provided. Connect credentials will take priority.",
	);
}
```

Reply:
Adjusted with [27cd56a](https://github.com/1Password/load-secrets-action/pull/36/commits/27cd56abb03b44c2c565eedae6c42a8446c6cde5).